src/routes/rooms.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import { AVAILABLE_ROOMS } from "../constants";
+import RoomReservation from "../models/RoomReservation";
+
+const RESERVATION_LENGTH = 1 * 60 * 60 * 1000; // 1 hour
+
+export async function getPublicRoomStatus(req: Request, res: Response) {
+  const room = AVAILABLE_ROOMS.find(r => r.id === req.query.id);
+
+  if (!room) {
+    return res.status(500).json({ message: "Room not found" });
+  }
+
+  const currentReservation = await RoomReservation.findOne({
+    room_id: req.query.id,
+    expiry: { $gte: Date.now() }
```

Comment:
What does this endpoint return if the room is expired?

Reply:
?

It just returns the room info unless I'm misunderstanding your question

Reply:
Oh oops I see